Automattic/jetpack
# Contact Form Block: Prevent removal of the submit button

Fixes #16303

#### Changes proposed in this Pull Request:

This PR aims to make the user experience better when using contact form blocks by removing the ability remove the submit button, since the button naturally is a required component of a contact form for this we: 
* add the attribute `lock { remove:  true}` to the submit button on each variation of a form. This makes it unable to remove the button for newly added forms.
* programatically find existing forms on a page and update the submit button attributes to prevent removal. This handles the case for users with existing forms.

#### Limitations:
The user can unlock the button and uncheck `prevent removal` and remove the button, but I think this might serve as an indicator to the user that this button shouldn't be removed. If they do remove the button they will find there's no other way to added it back and recreate the form at which point the button will be locked from removal again.

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?




#### Testing instructions:






1. Set up a Jurassic Ninja site.
2. Connect it to WP.com.
3. First create/save a post with a form block without these changes.
4. Apply changes then visit the post and verify that the option to remove the form button is no longer present.
5. Add a new form to the post and verify that the option to remove the button is no present.


**Before**
![Screenshot 2022-06-24 at 12 34 35 PM](https://user-images.githubusercontent.com/47489215/175613280-2bbdd5ea-b816-47f6-8955-960efe49d483.png)
**After**
![Screenshot 2022-06-24 at 12 36 12 PM](https://user-images.githubusercontent.com/47489215/175613323-a3d8e7f2-b34a-40df-9eb9-0cba1f299ef7.png)
